published_bucket.tf

```
@@ -407,3 +407,114 @@ resource "aws_iam_policy" "analytical_dataset_generator_read_write_non_pii" {
   description = "Allow read writing of non-pii data"
   policy      = data.aws_iam_policy_document.analytical_dataset_generator_read_write_non_pii.json
 }
+
+# policy for s3 read access of both non-pii and pii PDM data
+
+data "aws_iam_policy_document" "pdm_read_pii_and_non_pii" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      aws_s3_bucket.published.arn,
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:Get*",
+      "s3:List*",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.published.arn}/analytical-dataset/*",
```

```suggestion
      "${aws_s3_bucket.published.arn}/pdm-dataset/*",
```
we also need to add the non-pii bucket here